service/src/main/java/greencity/security/service/CustomOAuth2UserServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class CustomOAuth2UserServiceImpl extends OidcUserService implements CustomOAuth2UserService {
+
+    private final ModelMapper modelMapper;
+    private final UserRepo userRepo;
+    private final JwtTool jwtTool;
+    private final UserService userService;
+
+    @Override
+    public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2AuthenticationException {
+        OidcUser oidcUser = super.loadUser(userRequest);
+        return processOidcUser(oidcUser);
+    }
+
+    @Transactional
+    public OidcUser processOidcUser(OidcUser oidcUser) {
+        Map<String, Object> attributes = oidcUser.getAttributes();
+
+        String email = (String) attributes.get("email");
+        if (email == null || email.isEmpty()) {
+            log.error("Email is missing in OAuth2 user attributes");
+            throw new OAuth2AuthenticationException("Email not found from OAuth2 provider");
+        }
+
+        log.info("Processing OAuth2 login for email: {}", email);
+
+        Optional<User> existingUser = userRepo.findByEmail(email);
+        if (existingUser.isPresent()) {
+            log.info("User with email {} already exists", email);
+            return oidcUser;
+        }
+
+        String name = (String) attributes.get("name");
+        if (name == null || name.isEmpty()) {
+            log.warn("Name is missing in OAuth2 user attributes for email: {}", email);
+            name = "Unknown";
+        }
+
+        String refreshTokenKey = attributes.get("refresh_token_key") != null
+                ? attributes.get("refresh_token_key").toString()
+                : generateDefaultRefreshTokenKey();
+
+        User user = createUser(name, email, refreshTokenKey);
+        log.info("Created new user with email: {}", email);
+
+        return oidcUser;
+
+    }
+
+    private User createUser(String name, String email , String refreshTokenKey) {
+        User user = User.builder()
+                .name(name)
+                .firstName(name)
+                .email(email)
+                .dateOfRegistration(LocalDateTime.now())
+                .role(Role.ROLE_USER)
+                .refreshTokenKey(refreshTokenKey)
+                .lastActivityTime(LocalDateTime.now())
+                .userStatus(UserStatus.CREATED)
+                .emailNotification(EmailNotification.DISABLED)
+                .rating(AppConstant.DEFAULT_RATING)
+                .language(Language.builder()
+                        .id(modelMapper.map("en", Long.class))
+                        .build())
+                .build();
+        return userRepo.save(user);
```

Comment:
Also except save you should updateStatus. I mean make UserStatus.ACTIVATED.
Or make it in line 86. where you making status.CREATED.

I guess when the user registers using Google it really does not need to confirm email addresses as usual users.
I mean maybe it makes sense to set userStatus to ACTIVATED instead of CREATED.

Reply:
Done